helix-term/src/commands/mouse.rs

```diff
@@ -0,0 +1,283 @@
+        select_long_word_mouse, "Select a long word",
+        select_all_mouse, "Select whole document with mouse",
+        scroll_up_mouse, "Scroll view up with mouse line",
+        scroll_down_mouse, "Scroll view down with mouse line",
+        paste_primary_clipboard_before_mouse, "Paste primary clipboard before mouse",
+        yank_main_selection_to_primary_clipboard_mouse, "Yank Main selection to primary clipboard through mouse",
+        replace_selections_with_primary_clipboard_mouse, "Replace selection with primary clipboard through mouse",
+        add_breakpoint_mouse, "Add breakpoint with the mouse when clicking on the numbers on the left",
+        add_selection_mouse, "Add a selection at you mouse pointer",
+        go_to_definition_mouse, "Goto Definition with mouse",
+        code_action_mouse, "Perform code action with mouse",
+        dap_edit_condition_mouse, "Edit breakpoint condition on current line with mouse",
+        dap_edit_log_mouse, "Edit breakpoint log message on current line with mouse",
+    );
+}
+
+fn handle_selection_in_buffer(
+    cx: &mut Context,
+    evt: &MouseEvent,
+    ev: &mut EditorView,
+    callback: impl Fn(&mut Document, &ViewId, usize),
+) -> bool {
+    let editor = &mut cx.editor;
+
+    if let Some((pos, view_id)) = pos_and_view(editor, evt.row, evt.column, true) {
+        let prev_view_id = view!(editor).id;
+        let doc = doc_mut!(editor, &view!(editor, view_id).doc);
+
+        callback(doc, &view_id, pos);
+
+        if view_id != prev_view_id {
+            ev.clear_completion(editor);
+        }
+
+        editor.focus(view_id);
+        editor.ensure_cursor_in_view(view_id);
+
+        return true;
+    }
+    log::info!("false in return");
+    false
+}
+
+fn handle_main_button_mouse(cx: &mut Context, evt: &MouseEvent, ev: &mut EditorView) {
+    if !handle_selection_in_buffer(
+        cx,
+        evt,
+        ev,
+        |doc: &mut Document, view_id: &ViewId, pos: usize| {
+            doc.set_selection(*view_id, Selection::point(pos));
+        },
+    ) {
+        add_breakpoint_mouse(cx, evt, ev);
+    }
+}
+
+fn set_mouse_selection(cx: &mut Context, evt: &MouseEvent, ev: &mut EditorView) {
+    handle_selection_in_buffer(
+        cx,
+        evt,
+        ev,
+        |doc: &mut Document, view_id: &ViewId, pos: usize| {
+            doc.set_selection(*view_id, Selection::point(pos));
+        },
+    );
+}
+
+fn select_word_mouse(cx: &mut Context, _: &MouseEvent, _: &mut EditorView) {
+    move_prev_word_start(cx);
+    move_next_word_end(cx);
+}
+
+fn select_long_word_mouse(cx: &mut Context, _: &MouseEvent, _: &mut EditorView) {
+    move_prev_long_word_start(cx);
+    move_next_long_word_end(cx);
+}
+
+fn select_all_mouse(cx: &mut Context, _: &MouseEvent, _: &mut EditorView) {
+    super::select_all(cx)
+}
+
+fn scroll_mouse_impl(cx: &mut Context, evt: &MouseEvent, dir: Direction, _: &mut EditorView) {
+    let current_view = cx.editor.tree.focus;
```

Comment:
Should we check if a picker has the focus?

currently the mouse wheel does not scroll the code preview in the picker, but the code in the current buffer.